run_unittests.py

```diff
@@ -4702,6 +4702,37 @@ def test_link_language_linker(self):
         self.assertRegex(contents, r'build main(\.exe)?.*: c_LINKER')
         self.assertRegex(contents, r'build (lib|cyg)?mylib.*: c_LINKER')
 
+    def test_cross_file_constants(self):
+        filenames = []
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as crossfile:
```

Comment:
There are a bunch of caveats to using `NamedTemporaryFile` like this on windows. There's  a `temp_filename` helper in run_unittests.py that's safe to use on windows, although it does always delete the file, you could probably do something like:

```python
with temp_filename() as cfile1, temp_filename() as cfile2:
    with open(cfile1, 'w') as f:
        ...
    with open(cfile2, 'w') as f:
        ....
    parser = ...
    values = ...
self.assert ...
```

Reply:
There are many tests that use NamedTemporaryFile. The problem on Windows is you cannot read the file while it's open, so you cannot use it in a `with` clause, but that leaks temp files I guess.

Updated the test using your pattern.